File: srf-core/src/main/java/org/reggsoft/srfcore/persistance/entity/DataSourceVirtual.java
package org.reggsoft.srfcore.persistance.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "ds_virtual")
public class DataSourceVirtual {

    @Id
    @Column
    private Integer id;
    @Column
    private String sid;
    @Column
    private String name;
    @Column
    private Integer updatePeriod;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getSid() {
        return sid;
    }

    public void setSid(String sid) {
        this.sid = sid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getUpdatePeriod() {
        return updatePeriod;
    }

    public void setUpdatePeriod(Integer updatePeriod) {
        this.updatePeriod = updatePeriod;
    }
}

File: srf-core/src/main/kotlin/org/reggster/srfcore/domain/datasources/virtual/DataSourceVirtual.kt
package org.reggster.srfcore.domain.datasources.virtual

import org.reggster.srfcore.domain.datasources.DataSourceType
import org.reggster.srfcore.domain.datasources.ScadaDataSource
import javax.persistence.Entity

@Entity
class DataSourceVirtual(
    override var type: DataSourceType = DataSourceType.VIRTUAL
): ScadaDataSource()